[Bug]: ELBv2 Provider v1.5.0+ LBListener 'panic'

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Resource(s)

elbv2.aws.upbound.io/v1beta1, Kind=LBListener

### Resource MRs required to reproduce the bug

```
---
apiVersion: elbv2.aws.upbound.io/v1beta1
kind: LBListener
metadata:
  name: https
spec:
  deletionPolicy: Delete
  forProvider:
    region: us-east-1
    protocol: TCP
    port: 443
    defaultAction:
      - type: forward
        targetGroupArn: validTargetGroupArn
    loadBalancerArn: validLoadBalancerArn
```

### Steps to Reproduce

Deploy elbv2.aws.upbound.io/v1beta1 LB, LBTargetGroup and LBListener with v1.4.0 is successful.
Bump elbv2 Provider version to v1.5.0 or v1.6.0, an `Observed a panic: value is null` is logged

### What happened?

After bumping version from v1.4.0 to v1.5.0+, the following is logged for existing LBListener:

```
2024-06-10T18:41:57Z	DEBUG	provider-aws	Observing the external resource	{"uid": "838a6eb0-7742-4b8c-ab9f-2163e6a7290b", "name": "use1-sandbox-primary-internal-j6h86-6xnx2", "gvk": "elbv2.aws.upbound.io/v1beta1, Kind=LBListener"}
E0610 18:41:57.677193       1 runtime.go:77] Observed a panic: value is null
goroutine 3841 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x134f03a0, 0x1936c980})
	k8s.io/apimachinery@v0.29.4/pkg/util/runtime/runtime.go:75 +0x85
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:108 +0xb2
panic({0x134f03a0?, 0x1936c980?})
	runtime/panic.go:770 +0x132
github.com/hashicorp/go-cty/cty.Value.LengthInt({{{0x195976d8?, 0xc00bdb2290?}}, {0x0?, 0x0?}})
	github.com/hashicorp/go-cty@v1.4.1-0.20200723130312-85980079f637/cty/value_ops.go:989 +0x258
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbForwardActionOneOf({{{0x195976d8?, 0xc00bdb22e0?}}, {0x130d90a0?, 0xc00bce7d10?}}, 0x0, {{0xc00c7f38e0, 0x7}, 0x0, 0x0, 0x0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:1036 +0x1de
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbForwardAction(0xc00d1e8400, {0x16800240, 0xe}, 0x0, {{0xc00c7f38e0, 0x7}, 0x0, 0x0, 0x0, 0xc0052f3b30, ...}, ...)
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:1014 +0x22b
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbListenerActions(0xc00d1e8400, {0x16800240, 0xe}, {0xc00cce80f0?, 0x1, 0x41a085?})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:979 +0x2c5
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.resourceListenerRead({0x19596cf8, 0xc005270d20}, 0xc00d1e8400, {0x167734a0?, 0xc00271c000?})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:535 +0x5db
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Read.interceptedHandler[...].func9(0xc00d1e8400?, {0x167734a0?, 0xc00271c000})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x19596da0?, {0x19596da0?, 0xc00ce11490?}, 0xd?, {0x167734a0?, 0xc00271c000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:790 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001e25b20, {0x19596da0, 0xc00ce11490}, 0xc00198c820, {0x167734a0, 0xc00271c000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:1089 +0x529
github.com/crossplane/upjet/pkg/controller.(*terraformPluginSDKExternal).Observe(0xc00ba9ddc0, {0x19596da0, 0xc00ce11490}, {0x1969ab10, 0xc00cd0d508})
	github.com/crossplane/upjet@v1.4.0/pkg/controller/external_tfpluginsdk.go:472 +0x173
github.com/crossplane/upjet/pkg/controller.(*terraformPluginSDKAsyncExternal).Observe(0xc00d23c1e0, {0x19596da0, 0xc00ce11490}, {0x1969ab10, 0xc00cd0d508})
	github.com/crossplane/upjet@v1.4.0/pkg/controller/external_async_tfpluginsdk.go:127 +0xc5
github.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile(0xc00a9b2f00, {0x19596cf8, 0xc005085440}, {{{0x0, 0x0}, {0xc0070bc690, 0x29}}})
	github.com/crossplane/crossplane-runtime@v1.16.0-rc.2.0.20240510094504-3f697876fa57/pkg/reconciler/managed/reconciler.go:914 +0x2150
github.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile(0xc006636640, {0x19596cf8, 0xc005085440}, {{{0x0?, 0x5?}, {0xc0070bc690?, 0xc00cb99d10?}}})
	github.com/crossplane/crossplane-runtime@v1.16.0-rc.2.0.20240510094504-3f697876fa57/pkg/ratelimiter/reconciler.go:54 +0x151
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x195b60d8?, {0x19596cf8?, 0xc005085440?}, {{{0x0?, 0xb?}, {0xc0070bc690?, 0x0?}}})
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0065aee60, {0x19596d30, 0xc0013d3b80}, {0x15172980, 0xc00ce0f2e0})
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:316 +0x3bc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0065aee60, {0x19596d30, 0xc0013d3b80})
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:266 +0x1be
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 3650
	sigs.k8s.io/controller-runtime@v0.17.3/pkg/internal/controller/controller.go:223 +0x50c
```

When attempting to create a new Listener with elbv2 v1.5.0, Listeners are created correctly and associated with Load-Balancer, but the following status is eventually returned:

```
status:
  atProvider: {}
  conditions:
    - lastTransitionTime: '2024-06-10T18:52:32Z'
      reason: Creating
      status: 'False'
      type: Ready
    - lastTransitionTime: '2024-06-10T18:54:30Z'
      message: >-
        cannot patch the managed resource via server-side apply:
        LBListener.elbv2.aws.upbound.io
        "use1-sandbox-primary-internal-j6h86-nv9z8" is invalid: spec: Invalid
        value: "object": spec.forProvider.defaultAction is a required parameter
      reason: ReconcileError
      status: 'False'
      type: Synced
    - lastTransitionTime: '2024-06-10T18:54:30Z'
      message: >-
        async create failed: failed to create the resource: [{0 creating ELBv2
        Listener (): operation error Elastic Load Balancing v2: CreateListener,
        https response error StatusCode: 400, RequestID:
        677dd570-2c5f-4a16-9644-44ce704005f7, api error ValidationError: The
        input parameters provided are not supported  []}]
      reason: AsyncCreateFailure
      status: 'False'
      type: LastAsyncOperation
```

### Relevant Error Output Snippet

_No response_

### Crossplane Version

1.16.0

### Provider Version

1.5.0

### Kubernetes Version

_No response_

### Kubernetes Distribution

_No response_

### Additional Info

_No response_

I'm seeing the same thing with the following setup:

```
% kubectl get providers
NAME                          INSTALLED   HEALTHY   PACKAGE                                                    AGE
provider-aws-autoscaling      True        True      xpkg.upbound.io/upbound/provider-aws-autoscaling:v1.10.0   57d
provider-aws-ec2              True        True      xpkg.upbound.io/upbound/provider-aws-ec2:v1.10.0           58d
provider-aws-ecr              True        True      xpkg.upbound.io/upbound/provider-aws-ecr:v1.10.0           22d
provider-aws-elbv2            True        True      xpkg.upbound.io/upbound/provider-aws-elbv2:v1.10.0         44h
provider-aws-iam              True        True      xpkg.upbound.io/upbound/provider-aws-iam:v1.10.0           63d
provider-aws-kms              True        True      xpkg.upbound.io/upbound/provider-aws-kms:v1.10.0           22d
provider-aws-s3               True        True      xpkg.upbound.io/upbound/provider-aws-s3:v1.10.0            64d
upbound-provider-family-aws   True        True      xpkg.upbound.io/upbound/provider-family-aws:v1.10.0        64d
```

Here is the resource I create:

```yaml
apiVersion: elbv2.aws.upbound.io/v1beta2
kind: LBListener
metadata:
  labels:
    environment: dev
    name: dev-us-east-1-private-consul-nlb-tcp-8500
    region: us-east-1
    subnet-type: private
    vpc-name: dev-us-east-1
  name: dev-us-east-1-private-consul-nlb-tcp-8500
spec:
  forProvider:
    defaultAction:
    - targetGroupArnSelector:
        matchLabels:
          environment: dev
          name: dev-us-east-1-private-consul-nlb-tcp-8500
          region: us-east-1
          subnet-type: private
          vpc-name: dev-us-east-1
      type: forward
    loadBalancerArnSelector:
      matchLabels:
        environment: dev
        name: dev-us-east-1-private-consul-nlb
        region: us-east-1
        subnet-type: private
        vpc-name: dev-us-east-1
    port: 8500
    protocol: TCP
    region: us-east-1
```

As soon as that resource is applied, the elbv2 provider panics:

```
{"level":"info","ts":"2024-08-02T16:20:10Z","logger":"provider-aws","msg":"Beta feature enabled","flag":"EnableBetaManagementPolicies"}
panic: value is null

goroutine 31497 [running]:
github.com/hashicorp/go-cty/cty.Value.LengthInt({{{0x1c4c14b8?, 0xc0036bc0f0?}}, {0x0?, 0x0?}})
	github.com/hashicorp/go-cty@v1.4.1-0.20200723130312-85980079f637/cty/value_ops.go:989 +0x258
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbForwardActionOneOf({{{0x1c4c14b8?, 0xc0036bc110?}}, {0x15724ee0?, 0xc003583848?}}, 0x0, {{0xc0047809c0, 0x7}, 0x0, 0x0, 0x0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:1038 +0x1de
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbForwardAction(0xc003c39600, {0x1944aa1c, 0xe}, 0x0, {{0xc0047809c0, 0x7}, 0x0, 0x0, 0x0, 0xc004c855c0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:1016 +0x22b
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.flattenLbListenerActions(0xc003c39600, {0x1944aa1c, 0xe}, {0xc0011a66e0?, 0x1, 0xc00abb58ad?})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:981 +0x2c5
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.resourceListenerRead({0x1c4c0ad8, 0xc007773b60}, 0xc003c39600, {0x193bb340?, 0xc00411d6c0?})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:537 +0x5db
github.com/hashicorp/terraform-provider-aws/internal/service/elbv2.resourceListenerCreate({0x1c4c0ad8, 0xc007773b60}, 0xc003c39600, {0x193bb340, 0xc00411d6c0})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/service/elbv2/listener.go:508 +0x11f3
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc003c39600?, {0x193bb340?, 0xc00411d6c0})
	github.com/hashicorp/terraform-provider-aws@v0.0.0-00010101000000-000000000000/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1c4c0b80?, {0x1c4c0b80?, 0xc00380b650?}, 0xd?, {0x193bb340?, 0xc00411d6c0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00207e460, {0x1c4c0b80, 0xc00380b650}, 0x0, 0xc000311900, {0x193bb340, 0xc00411d6c0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/crossplane/upjet/pkg/controller.(*terraformPluginSDKExternal).Create(0xc0045c0c40, {0x1c4c0b80, 0xc00380b650}, {0x1c5f4a60, 0xc002d82f08})
	github.com/crossplane/upjet@v1.4.1-0.20240612123927-37c7f4e91d57/pkg/controller/external_tfpluginsdk.go:592 +0xdd
github.com/crossplane/upjet/pkg/controller.(*terraformPluginSDKAsyncExternal).Create.func1()
	github.com/crossplane/upjet@v1.4.1-0.20240612123927-37c7f4e91d57/pkg/controller/external_async_tfpluginsdk.go:149 +0x145
created by github.com/crossplane/upjet/pkg/controller.(*terraformPluginSDKAsyncExternal).Create in goroutine 3779
	github.com/crossplane/upjet@v1.4.1-0.20240612123927-37c7f4e91d57/pkg/controller/external_async_tfpluginsdk.go:145 +0x14c
```

Once it panics, I can't actually describe the resource because the webhook isn't running anymore. The listener _does_ get created on the AWS side though, so I think this isn't a bug on terraform but on whatever is processing the response from terraform - maybe not understanding that a field is nullable if the action is a forward action?

Note that the first example on the docs [here](https://marketplace.upbound.io/providers/upbound/provider-aws-elbv2/v1.10.0/resources/elbv2.aws.upbound.io/LBListener/v1beta1) is also broken - if you omit the `protocol`, crossplane won't apply the resource - but then adding the protocol (as shown in the last example) fails with the aforementioned panic.

--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ PROJECT_REPO := github.com/upbound/$(PROJECT_NAME)
 
 export TERRAFORM_VERSION := 1.5.5
 export TERRAFORM_PROVIDER_VERSION := 5.68.0
-export TERRAFORM_PROVIDER_RELEASE := v$(TERRAFORM_PROVIDER_VERSION)-upjet.1
+export TERRAFORM_PROVIDER_RELEASE := v$(TERRAFORM_PROVIDER_VERSION)-upjet.2
 export TERRAFORM_PROVIDER_SOURCE := hashicorp/aws
 export TERRAFORM_PROVIDER_REPO ?= https://github.com/hashicorp/terraform-provider-aws
 export TERRAFORM_DOCS_PATH ?= website/docs/r

--- a/config/elbv2/config.go
+++ b/config/elbv2/config.go
@@ -4,7 +4,14 @@
 
 package elbv2
 
-import "github.com/crossplane/upjet/pkg/config"
+import (
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/crossplane/upjet/pkg/config"
+)
 
 // Configure adds configurations for the elbv2 group.
 func Configure(p *config.Provider) {
@@ -43,6 +50,64 @@ func Configure(p *config.Provider) {
 			"default_action.forward.target_group.arn": {
 				TerraformName: "aws_lb_target_group",
 			},
+		}
+
+		// lb_listener schema allows to configure "default_action" with type
+		// "forward", in 2 different ways.
+		// 1. you can specify, default_action.0.forward, which allows configuring
+		//    multiple target groups.
+		// 2. you can specify default_action.0.target_group_arn if you want
+		//    to configure only one target group.
+		// Former is a more general way, and latter is more of a shortcut for
+		// a specific case, which can already be expressed with #1.
+		// TF implementation instructs to specify either #1 or #2, not both.
+		// However, they both end up in the state redundantly and cause
+		// unnecessary diff.
+		r.TerraformCustomDiff = func(diff *terraform.InstanceDiff, _ *terraform.InstanceState, _ *terraform.ResourceConfig) (*terraform.InstanceDiff, error) { //nolint:gocyclo
+			// skip no diff or destroy diffs
+			if diff == nil || diff.Empty() || diff.Destroy || diff.Attributes == nil {
+				return diff, nil
+			}
+			duplicatedAction := ""
+			for k, attrDiff := range diff.Attributes {
+				// user specified the "default_action.0.target_group_arn" and
+				// "default_action.0.forward" is not specified in config.
+				// In that case, default_action.0.forward is populated
+				// by the AWS API, which can cause an unnecessary diff,
+				// trying to remove that "auto-populated" element.
+				if regexp.MustCompile(`^default_action\.\d+\.forward\.#$`).MatchString(k) &&
+					attrDiff.New == "0" && attrDiff.Old == "1" {
+					delete(diff.Attributes, k)
+					// save this attribute path to remove remaining diffs
+					// of its nested fields if any in a second pass
+					duplicatedAction = strings.TrimSuffix(k, ".#")
+				}
+				// this is the same case as above
+				if regexp.MustCompile(`^default_action\.\d+\.forward\.\d+\.target_group\.#$`).MatchString(k) &&
+					attrDiff.New == "0" && attrDiff.Old == "1" {
+					delete(diff.Attributes, k)
+				}
+				// this is the same case but vice versa. user specified
+				// forward target via default_action.0.forward, and
+				// default_action.0.target_group_arn is omitted.
+				// In that case, default_action.0.target_group_arn is populated
+				// by the AWS API and ends up in the state, causing
+				// an unnecessary diff, which we omit here.
+				if regexp.MustCompile(`^default_action\.\d+\.target_group_arn$`).MatchString(k) &&
+					attrDiff.New == "" && attrDiff.Old != "" && attrDiff.NewRemoved {
+					delete(diff.Attributes, k)
+				}
+			}
+			// if we have caught an unnecessary diff for default_action.0.forward, remove
+			// any sub-element diffs of it.
+			if duplicatedAction != "" {
+				for k := range diff.Attributes {
+					if strings.HasPrefix(k, duplicatedAction) {
+						delete(diff.Attributes, k)
+					}
+				}
+			}
+			return diff, nil
 		}
 	})
 

--- a/config/elbv2/config.go
+++ b/config/elbv2/config.go
@@ -14,7 +14,7 @@ import (
 )
 
 // Configure adds configurations for the elbv2 group.
-func Configure(p *config.Provider) {
+func Configure(p *config.Provider) { //nolint:gocyclo
 	p.AddResourceConfigurator("aws_lb", func(r *config.Resource) {
 		r.ExternalName.OmittedFields = append(r.ExternalName.OmittedFields, "name_prefix")
 		r.References = config.References{

--- a/config/elbv2/config.go
+++ b/config/elbv2/config.go
@@ -119,6 +119,19 @@ func Configure(p *config.Provider) { //nolint:gocyclo
 			s.Computed = false
 		}
 		r.LateInitializer.IgnoredFields = []string{"target_failover"}
+		r.TerraformCustomDiff = func(diff *terraform.InstanceDiff, _ *terraform.InstanceState, _ *terraform.ResourceConfig) (*terraform.InstanceDiff, error) {
+			// skip no diff or destroy diffs
+			if diff == nil || diff.Empty() || diff.Destroy || diff.Attributes == nil {
+				return diff, nil
+			}
+
+			// ignore diff due to defaulting in the TF schema
+			udiDiff, ok := diff.Attributes["target_health_state.0.unhealthy_draining_interval"]
+			if ok && udiDiff.Old == "" && udiDiff.New == "0" {
+				delete(diff.Attributes, "target_health_state.0.unhealthy_draining_interval")
+			}
+			return diff, nil
+		}
 	})
 
 	p.AddResourceConfigurator("aws_lb_target_group_attachment", func(r *config.Resource) {

--- a/examples/elbv2/v1beta2/lblistener-forward-multiple-targetgroups.yaml
+++ b/examples/elbv2/v1beta2/lblistener-forward-multiple-targetgroups.yaml
@@ -1,0 +1,195 @@
+# SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LBListener
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: front-end
+spec:
+  forProvider:
+    defaultAction:
+    - type: forward
+      forward:
+        targetGroup:
+          - arnSelector:
+              matchLabels:
+                testing.upbound.io/example-name: lblistener-forward-multiple
+          - arnSelector:
+              matchLabels:
+                testing.upbound.io/example-name: lblistener-forward-multiple-2
+        stickiness:
+          enabled: true
+          duration: 3600
+    loadBalancerArnSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+    port: 80
+    protocol: HTTP
+    region: us-west-1
+---
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LBTargetGroup
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple-2
+  name: test-2
+spec:
+  forProvider:
+    name: example-lb-tg-2
+    port: 8080
+    protocol: HTTP
+    region: us-west-1
+    targetType: ip
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+
+---
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LBTargetGroup
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: test
+spec:
+  forProvider:
+    name: example-lb-tg
+    port: 80
+    protocol: HTTP
+    region: us-west-1
+    targetType: ip
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+
+---
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LB
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: test
+spec:
+  forProvider:
+    enableDeletionProtection: false
+    internal: false
+    loadBalancerType: application
+    name: test-lb-tf
+    region: us-west-1
+    securityGroupSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+    subnetSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+    tags:
+      Environment: production
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: VPC
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: elbv2-vpc
+spec:
+  forProvider:
+    cidrBlock: 172.16.0.0/16
+    region: us-west-1
+    tags:
+      Name: DemoVpc
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: InternetGateway
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: elbv2-internet-gateway
+spec:
+  forProvider:
+    region: us-west-1
+    tags:
+      Name: testing
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: elbv2-subnet
+spec:
+  forProvider:
+    availabilityZone: us-west-1a
+    cidrBlock: 172.16.10.0/24
+    region: us-west-1
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: elbv2-subnet-2
+spec:
+  forProvider:
+    availabilityZone: us-west-1b
+    cidrBlock: 172.16.20.0/24
+    region: us-west-1
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: SecurityGroup
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-multiple
+  name: elbv2-securitygroup
+spec:
+  forProvider:
+    description: Allow TLS inbound traffic
+    name: allow_tls
+    region: us-west-1
+    tags:
+      Name: allow_tls
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-multiple

--- a/examples/elbv2/v1beta2/lblistener-forward-single-targetgroup.yaml
+++ b/examples/elbv2/v1beta2/lblistener-forward-single-targetgroup.yaml
@@ -1,0 +1,170 @@
+# SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LBListener
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: front-end
+spec:
+  forProvider:
+    defaultAction:
+    - type: forward
+      forward:
+        targetGroup:
+          - arnSelector:
+              matchLabels:
+                testing.upbound.io/example-name: lblistener-forward-single
+
+    loadBalancerArnSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+    port: 80
+    protocol: HTTP
+    region: us-west-1
+
+---
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LBTargetGroup
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: test
+spec:
+  forProvider:
+    name: example-lb-tg
+    port: 80
+    protocol: HTTP
+    region: us-west-1
+    targetType: ip
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+
+---
+
+apiVersion: elbv2.aws.upbound.io/v1beta2
+kind: LB
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: test
+spec:
+  forProvider:
+    enableDeletionProtection: false
+    internal: false
+    loadBalancerType: application
+    name: test-lb-tf
+    region: us-west-1
+    securityGroupSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+    subnetSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+    tags:
+      Environment: production
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: VPC
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: elbv2-vpc
+spec:
+  forProvider:
+    cidrBlock: 172.16.0.0/16
+    region: us-west-1
+    tags:
+      Name: DemoVpc
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: InternetGateway
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: elbv2-internet-gateway
+spec:
+  forProvider:
+    region: us-west-1
+    tags:
+      Name: testing
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: elbv2-subnet
+spec:
+  forProvider:
+    availabilityZone: us-west-1a
+    cidrBlock: 172.16.10.0/24
+    region: us-west-1
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: elbv2-subnet-2
+spec:
+  forProvider:
+    availabilityZone: us-west-1b
+    cidrBlock: 172.16.20.0/24
+    region: us-west-1
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: SecurityGroup
+metadata:
+  annotations:
+    meta.upbound.io/example-id: elbv2/v1beta2/lblistener
+  labels:
+    testing.upbound.io/example-name: lblistener-forward-single
+  name: elbv2-securitygroup
+spec:
+  forProvider:
+    description: Allow TLS inbound traffic
+    name: allow_tls
+    region: us-west-1
+    tags:
+      Name: allow_tls
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: lblistener-forward-single

--- a/go.mod
+++ b/go.mod
@@ -435,4 +435,4 @@ require (
 // copied from the Terraform provider
 replace github.com/hashicorp/terraform-plugin-log => github.com/gdavison/terraform-plugin-log v0.0.0-20230928191232-6c653d8ef8fb
 
-replace github.com/hashicorp/terraform-provider-aws => github.com/upbound/terraform-provider-aws v0.0.0-20241008123507-c58f3dd551f5
+replace github.com/hashicorp/terraform-provider-aws => github.com/upbound/terraform-provider-aws v0.0.0-20241019052313-8fed8454b0e3

